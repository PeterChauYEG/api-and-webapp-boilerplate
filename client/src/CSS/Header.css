.Header {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  color: black;
  text-align: center;
  padding: 8px 8px;
}
