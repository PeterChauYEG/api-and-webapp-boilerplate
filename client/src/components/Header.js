import React from 'react'

// import css
import '../CSS/Header.css'

const Header = (props) => {
  const {
    backgroundColor,
  } = props

  return (
    <div className={`Header ${backgroundColor}`}>
      {props.children}
    </div>
  );
}

Header.propTypes = {
  backgroundColor: React.PropTypes.string.isRequired,
}

export default Header;
