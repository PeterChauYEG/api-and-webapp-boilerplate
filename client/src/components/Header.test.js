import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import Header from './Header'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const backgroundColor = ""
  const children = <p>{process.env.NODE_ENV}</p>
  
  // render component
  const component = renderer.create(
    <Header backgroundColor={backgroundColor}>
      {children}
    </Header>
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});
