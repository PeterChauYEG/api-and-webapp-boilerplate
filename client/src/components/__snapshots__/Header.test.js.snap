exports[`test renders without crashing 1`] = `
<div
  className="Header ">
  <p>
    test
  </p>
</div>
`;
